### DISCUSSION TITLE
Thinlens Sensor: Scene units and aperture radius

### DISCUSSION THREAD

**Discussion by maxfrei750** [mitsuba-renderer/mitsuba3#724] [Q&A]
I'm trying to export a camera with depth-of-field from Blender to Mitsuba (if everything goes well, then this might even be added to the [Mitsuba Blender Add-on](https://github.com/mitsuba-renderer/mitsuba-blender)).

To get DOF in Mitsuba, I need to use the [`thinslens`](https://mitsuba.readthedocs.io/en/stable/src/generated/plugins_sensors.html#perspective-camera-with-a-thin-lens-thinlens) sensor, which (among other parameters) expects a `focal_length` and an `aperture_radius`. While Blender specifies the focal length (in mm, either directly or via the field of view), the aperture radius is defined via the f-stop property. According to [Wikipedia](https://en.wikipedia.org/wiki/Aperture#Aperture_area), I can calculate the aperture radius from the focal length and the f-number. 

Naturally, in the real-world, the aperture radius will be in units of length. However, the Mitsuba documentation of the `aperture_radius` states that it has to be specified in scene units:

> aperture_radius | float | Denotes the radius of the camera’s aperture in **scene units**.

What is the best way to tackle this problem?

Thanks!

**Comment by njroussel** [comment]
Hi @maxfrei750 

The `focal_length` parameter of the `thinlens` plugin is also only used to compute the **diagonal** field of view. Here's the relevant source code: https://github.com/mitsuba-renderer/mitsuba3/blob/master/src/render/sensor.cpp#L167-L181
It's given in `mm` because it's _relative_ to the `35 mm` film. In reality, it doesn't really matter, as it's only used to compute the field of view. Note that this also explains why only one of `fov` and `focal_length` should be provided.

You can arbitrarily set the scale of your scene relative to scene units (1 mm == 1 scene unit for example). Hence, the aperture radius being defined in scene units. The f-stop measure is a relative measure, you only need the focal length and aperture diameter to be in the same units.

Does this make sense?
